vpr/src/device/rr_graph_view.h

```diff
@@ -397,7 +400,12 @@ class RRGraphView {
     inline const t_rr_switch_inf& rr_switch_inf(RRSwitchId switch_id) const {
         return rr_switch_inf_[switch_id];
     }
-
+    inline size_t num_rr_switches() const {
+        return rr_switch_inf_.size();
+    }
+    const vtr::vector<RRSwitchId, t_rr_switch_inf>& rr_switch() const {
```

Comment:
No need to return a read-only vector here. You can remove the ``const`` and ``&``